### PR TITLE
PPS Patients list search

### PR DESCRIPTION
**Pull request by davidbgomes**
### :pushpin: References

-   **Issue:** Closes [Add patient search functionality](https://app.clickup.com/t/862kkau88)

### :memo: Implementation

- Create input textbox to search patients by ID or Code

### :video_camera: Screenshots/Screen capture
image

### CODE REVIEW

File: src/data/repositories/SurveyFormD2Repository.ts [hunk 1]
```diff
@@ -44,6 +44,8 @@ const SURVEY_NAME_DATAELEMENT_ID = "mEQnAQQjdO8";
 const SURVEY_HOSPITAL_CODE_DATAELEMENT_ID = "uAe6Mlw2XlE";
 const SURVEY_WARD_CODE_DATAELEMENT_ID = "q4mg5z04dzd";
 const SURVEY_PATIENT_CODE_DATAELEMENT_ID = "yScrOW1eTvm";
+const PATIENT_ID = "eKUbBRlR684";
+const PATIENT_CODE = "yScrOW1eTvm";
```
Comment (9sneha-n) [inline src/data/repositories/SurveyFormD2Repository.ts:48]: reuse existing SURVEY_PATIENT_CODE_DATAELEMENT_ID for PATIENT_CODE (not a blocker, can address in next PR)

File: src/webapp/components/survey-list/SurveyList.tsx [hunk 2]
```diff
@@ -99,12 +111,17 @@ export const SurveyList: React.FC<SurveyListProps> = ({ surveyFormType }) => {
...
                     )}
                     <SurveyListTable
-                        surveys={filteredSurveys}
+                        surveys={surveyList}
                         surveyFormType={surveyFormType}
                         updateSelectedSurveyDetails={updateSelectedSurveyDetails}
                     />
                 </CustomCard>
             </ContentLoader>
+            {isLoading && (
```
Comment (9sneha-n) [inline src/webapp/components/survey-list/SurveyList.tsx:120]: use content loader for loading component, feel free to customize the component

File: src/webapp/components/survey-list/SurveyList.tsx [hunk 3]
```diff
@@ -74,6 +77,15 @@ export const SurveyList: React.FC<SurveyListProps> = ({ surveyFormType }) => {
                         surveys
                     ) && (
                         <ButtonWrapper>
+                            {surveyFormType === "PPSPatientRegister" && (
+                                <TextField
+                                    label={i18n.t("Search Patient")}
+                                    helperText={i18n.t("Filter by patient id or code")}
+                                    value={patientFilterKeyword}
+                                    onChange={e => setPatientFilterKeyword(e.target.value)}
```
Comment (9sneha-n) [inline src/webapp/components/survey-list/SurveyList.tsx:85]: if setPatientFilterKeyword is done when the form is of PPSPatientRegister type, then the user goes to another form type, the patientFilterKeyword  will still be set. Correct? Can you test this? I think we need to reset patientFilterKeyword to empty whenever the formtype is not PPSPatientRegister.

File: src/webapp/components/survey-list/hook/useSurveyFilters.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }, [filteredSurveys]);
+
+    useEffect(() => {
+        if (!patientFilterKeyword && filteredSurveys) {
+            setSurveyList(filteredSurveys);
+        }
+    }, [filteredSurveys, patientFilterKeyword]);
+
+    const handleKeyPress = (event: React.KeyboardEvent<HTMLDivElement>) => {
+        if (patientFilterKeyword && event.key === "Enter") {
```
Comment (9sneha-n) [inline src/webapp/components/survey-list/hook/useSurveyFilters.ts:28]: Can we add an additonal check here, if surveyFormType === PPSPatientRegister for safety ?

File: src/webapp/components/survey-list/hook/useSurveyFilters.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                    () => {
+                        snackbar.error(i18n.t("Error fetching surveys"));
+                        setIsLoading(false);
+                    }
+                );
+        }
+    };
+
+    return {
+        surveyList,
```
Comment (9sneha-n) [inline src/webapp/components/survey-list/hook/useSurveyFilters.ts:46]: Can we rename this surveyList --> filteredPatientList and the custome hook useSurveyFilters --> usePatientSurveyFilter so we can consider this very specific to patient. We can change it in the future to more generic if required. That way all the conditions will make sense.

**Review by 9sneha-n — CHANGES_REQUESTED** [review]
Need to ensure that this code is applied only on PPSPatientRegister form type

File: src/webapp/components/survey-list/SurveyList.tsx (hunk 3 above)
Comment (davidbgomes) [reply src/webapp/components/survey-list/SurveyList.tsx:85]: I've tested and every time a user enters a new survey with some keyword entered, it is reset on render, so I think it should be fine.

**Comment by davidbgomes** [comment]
Thank you @9sneha-n ! I've added some comments to your comments, for whenever you can check them :) I'll make the changes we agree on in the meantime.

File: src/data/repositories/UserD2Repository.ts [hunk 6]
```diff
@@ -126,6 +126,8 @@ export class UserD2Repository implements UserRepository {
                         id: true,
                     },
                 },
+                userOnly: true,
```
Comment (9sneha-n) [inline src/data/repositories/UserD2Repository.ts:129]: @davidbgomes Incorrect code checked in, please revert

**Pull request merged**